dev/doxygen/pages/release_workflow.md

```diff
@@ -0,0 +1,155 @@
+a breaking change, which is a file format change in our case (we consider *any*
+file format change as breaking because we don't provide [forward compatibility]
+at all).
+
+As the file format also needs a version number, we just use the application's
+major version as the file format version. For example the application version
+"1.2" has the file format "1", and "13.37.42" has file format "13". Please note
+the special case of 0.x versions where the second number denotes the major
+version, so the application version "0.1.2" has file format "0.1". This way, the
+file format can always be determined by the application's version number.
+
+Given that versioning system, it's clear that any application with the same
+major version uses exactly the same file format. And for developers it's clear
+that when changing something in the file format, the application's major version
+*must* be incremented.
+
+
+# Preprocessor Defines {#doc_release_workflow_defines}
+
+The application version is compiled into the binary with the preprocessor define
+`APP_VERSION` and the file format version with the define `FILE_FORMAT_VERSION`.
+Whether the file format is stable or not is controlled by the define
+`FILE_FORMAT_STABLE`. See following chapters for details.
+
+
+# Git Branches {#doc_release_workflow_branches}
+
+The continuous development happens on feature/bugfix branches (or pull requests),
+which are merged into the `master` branch as soon as they are finished. As new
+features sometimes also introduce file format changes, the `master` branch must
+always be considered to contain the (unstable) *next* major version of the
+application. For example if the last stable release was version "2.3.4", the
+`master` must immediately be considered as version "3.0.0-unstable".
+
+This also means that on the `master` branch we cannot provide updates for
+already released versions. So we have to provide such updates in release
+branches. For every major version we create a release branch, which is then used
+for *any* update for that major version. For example on the branch `release/2`
+we would provide the releases "2.0", "2.0.1" (bugfix) and "2.1.0" (new feature).
+
+![Git Branches](release_workflow_diagram.png)
+
+Features and bugfixes which do not affect the file format may be cherry-picked
+from `master` to release branches. This way we can publish updates for the last
+officially released application version while the next major version is still in
+development. Generally we do *not* maintain older releases than the last stable
+release because this would be too much effort. So for example as soon as the
+version "0.2.0" is released, the version "0.1.x" is no longer maintained.
+
+
+# Git Tags
+
+For every officially released version we create a Git tag on the corresponding
+release branch. The tag name contains all three numbers of the application's
+version number. Examples: "0.1.0-rc1", "0.1.0", "1.2.0", "1.2.1"
```

Comment:
I'd propose that we PGP-sign all release tags and the version bump commits using a PGP key that we publish on our website and maybe on https://keybase.io/.

The same PGP key would be used for signing releases.

Reply:
Yeah for tags and releases it would be nice. But why the version bump commits?

Reply:
Mainly because - in contrast to tags - the commit signature is rendered/displayed on GitHub :slightly_smiling_face: Also, because commits are harder to manipulate than commits.

Reply:
But how is a version bump commit different than any other commit?

> Also, because commits are harder to manipulate than commits.

:grin:

Reply:
Ah, probably you mean the version bump commits on the release branches, not on master, and these commits are shown on GitHub because tags/branches point to them :)

Reply:
Signatures are shown for any signed commits. I only mentioned bump commits because I think it's too tedious to sign every commit, and the version bump commit indirectly signs the entire history chain that leads up to it :slightly_smiling_face:

Reply:
> I only mentioned bump commits because I think it's too tedious to sign every commit, and the version bump commit indirectly signs the entire history chain that leads up to it

Hmm but the signed tag does exactly the same?!

Reply:
Uhm, didn't you ask this question already?

> But why the version bump commits?

> Mainly because - in contrast to tags - the commit signature is rendered/displayed on GitHub :slightly_smiling_face:  Also, because commits are harder to manipulate than commits.

But it was just a suggestion, do whatever you want as long as something's signed.